src/FormBuilder/resources/js/form-builder/src/components/DatePicker/index.tsx

```diff
@@ -0,0 +1,136 @@
+import {__} from '@wordpress/i18n';
+import {useRef, useState} from 'react';
+import {close} from '@wordpress/icons';
+import {Button, DatePicker, DateTimePicker, PanelRow, Popover, TextControl} from '@wordpress/components';
+
+import './styles.scss';
+
+interface DatePickerProps {
+    label: string;
+    placeholder: string;
+    date: string;
+    onSelect: (date: string) => void;
+    invalidDateBefore?: string;
+    invalidDateAfter?: string;
+    is12Hour?: boolean;
+    showTimeSelector?: boolean;
+}
+
+/**
+ * @unreleased
+ */
+export default ({
+    label,
+    placeholder,
+    date: value,
+    onSelect,
+    invalidDateBefore = null,
+    invalidDateAfter = null,
+    is12Hour = true,
+    showTimeSelector = false,
+}: DatePickerProps) => {
+
+    const popoverRef = useRef();
+
+    const [date, setDate] = useState<string>(value);
+    const [isVisible, setIsVisible] = useState<boolean>(false);
+    const currentDate =  date ? new Date(date) : new Date();
+
+
+    const convertJsDateToMySQLDate = (dateTime: string) => {
+        // split the ISO string into date and time
+        const [date, time] = new Date(dateTime).toISOString().split('T');
+
+        return `${date} ${time.slice(0, 8)}`;
+    };
+
+    const toggleVisible = () => {
+        setIsVisible((state) => !state);
+    };
+
+    const onSelectDate = () => {
+        onSelect(convertJsDateToMySQLDate(date));
+        setIsVisible(false);
+    };
+
+    const checkDate = (date: Date) => {
+        if (invalidDateBefore) {
+            return date < new Date(invalidDateBefore);
+        }
+
+        if (invalidDateAfter) {
```

Comment:
Is there a situation where both `invalidDateBefore` and `invalidDateAfter` have values? What would be the expected outcome?

Reply:
Setting both parameters will give you a selectable date range, and the user can only select dates within that range. I added this to prevent users from selecting an end date that is before the start date.

Reply:
Ok, I think I misunderstood how this was being used.